src/js/view/controls/components/settings/menu.js

```diff
@@ -19,22 +19,51 @@ export function SettingsMenu(onVisibility, onSubmenuAdded, onMenuEmpty) {
 
     const settingsMenuElement = createElement(SettingsMenuTemplate());
 
-    const closeOnEnter = function(evt) {
-        if (evt && evt.keyCode === 27) {
-            instance.close(evt);
+    const handleKeyDown = function(evt) {
+        if (evt) {
+            switch (evt.keyCode) {
+                case 27: // esc
+                    instance.close();
+                    break;
+                case 37: // left-arrow
+                    if (evt.target.previousElementSibling) {
+                        evt.target.previousElementSibling.focus();
+                    } else {
+                        instance.close();
+                    }
+                    break;
+                case 38: // up-arrow
+                    instance.activateSubmenu(evt.target.getAttribute('name'), true);
+                    break;
+                case 39: // right-arrow
+                    if (evt.target !== closeButton.element()) {
+                        evt.target.nextElementSibling.focus();
+                    }
+                    break;
+                case 40: // down-arrow
+                    instance.activateSubmenu(evt.target.getAttribute('name'));
```

Comment:
What's the difference between up and down arrows? You could make the code dryer like this:
```js
  case 38: // up-arrow
  case 40: // down-arrow
    instance.activateSubmenu(evt.target.getAttribute('name'), evt.keyCode === 38);
    break;
```